nsv13/code/controllers/subsystem/overmap_mode.dm

```diff
@@ -387,29 +388,22 @@ SUBSYSTEM_DEF(overmap_mode)
 	hard_mode_enabled = !hard_mode_enabled
 	log_game("Hard mode has been [hard_mode_enabled ? "enabled" : "disabled"]!")
 	if(hard_mode_enabled) //Go on, clear the map out
+		force_mode(/datum/overmap_gamemode/hardmode)
```

Comment:
Currently, this will still runtime if done before roundstart, see `force_mode()` comment.